### PR TITLE
Definitions for ahocorasick

### PR DESCRIPTION
**Pull request by Longqin88888**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:

- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:

- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the `package.json`.

If removing a declaration:

- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

File: types/ahocorasick/index.d.ts
Comment (codershiba) [inline types/ahocorasick/index.d.ts:4]:
```suggestion
export class AhoCorasick {
```

Comment (e6nlaq) [reply types/ahocorasick/index.d.ts:4]: @codershiba Shouldn't it be correctly written as follows?

```diff typescript
- export default class AhoCorasick {
+ declare class AhoCorasick {
  // ...
  }
+ export = AhoCorasick;
```

File: types/ahocorasick/index.d.ts
Comment (e6nlaq) [inline types/ahocorasick/index.d.ts:7]: A more accurate type can be used for this part.

```suggestion
    search(match: string): [number, string][]
```

File: types/ahocorasick/index.d.ts
Comment (codershiba) [inline types/ahocorasick/index.d.ts:4]:
```suggestion
declare class AhoCorasick {
```

File: types/ahocorasick/index.d.ts
Comment (codershiba) [inline types/ahocorasick/index.d.ts:8]:
```suggestion
}

export = AhoCorasick;
```

File: types/ahocorasick/index.d.ts
Comment (codershiba) [inline types/ahocorasick/index.d.ts:7]:
```suggestion
    search(match: string): Array<number, string[]>;
```

Comment (e6nlaq) [reply types/ahocorasick/index.d.ts:7]: @codershiba Thanks for fixing it! But the correct type is `Array<[number, string[]]>`.

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/ahocorasick/package.json
1. types/ahocorasick/index.d.ts
1. types/ahocorasick/ahocorasick-tests.ts

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 744c848cddf9ef0bad6dd0f8765e82c0f7a40591
</p>

File: types/ahocorasick/index.d.ts
Comment (codershiba) [reply types/ahocorasick/index.d.ts]: Yeah, missed that but seems to be fixed by the author now.

**Review by ghost — APPROVED** [review]

**Review by e6nlaq — APPROVED** [review]
LGTM!
Final decision will be made by the DT maintainer.

File: types/ahocorasick/index.d.ts
Comment (sheetalkamat) [inline types/ahocorasick/index.d.ts:10]:
```suggestion
export = AhoCorasick;
export as namespace AhoCorasick;
```

Comment (sheetalkamat) [reply types/ahocorasick/index.d.ts:10]: For more details https://github.com/DefinitelyTyped/DefinitelyTyped?tab=readme-ov-file#how-do-i-write-definitions-for-packages-that-can-be-used-globally-and-as-a-module

File: types/ahocorasick/tsconfig.json
Comment (sheetalkamat) [inline types/ahocorasick/tsconfig.json:17]: Add browser test that uses this as global. https://github.com/DefinitelyTyped/DefinitelyTyped?tab=readme-ov-file#how-do-i-write-definitions-for-packages-that-can-be-used-globally-and-as-a-module

**Review by sheetalkamat — CHANGES_REQUESTED** [review]

**Review by sheetalkamat — CHANGES_REQUESTED** [review]
Also pls format using ` pnpm dprint fmt`

File: types/ahocorasick/index.d.ts
Comment (Longqin88888) [reply types/ahocorasick/index.d.ts]: Completed

Comment (Longqin88888) [reply types/ahocorasick/index.d.ts]: After adding this modification, the access control will report an error

**Comment by Longqin88888** [comment]
Can this pr be combined?

**Review by RyanCavanaugh — APPROVED** [review]

**Comment by Longqin88888** [comment]
Ready to merge

**Pull request merged**
